### PR TITLE
[RL] RL support varlen flash mask

### PR DESCRIPTION
**Pull request by JunnYu**
#### Before submitting

- [x] Lint code. If there are lint issues, please format the code first.

```shell
# Install and register `pre-commit` in the project folder
pip install pre-commit && pre-commit install

# Process previous code files separately
pre-commit run --file XXXX.py
```

- [x] Add test cases into `tests` folder. If there are codecov issues, please add tests cases first.

### PR types

New features
### PR changes

APIs
### Description

1. RL 支持varlen的flash mask，当开启use_remove_padding后，可以去掉batch里面的padding。
2. rollout_max_num_seqs新增参数，rollout的推理服务的最大并发（continue batching会使用）
3. quant_type增加前缀明确为rollout部分的rollout_quant_type
4. 默认不使用fused_rope
5. continue batching的时候支持传入repeat_num
6. 修复qwen2模型的rope的bug

### CODE REVIEW

File: paddlenlp/rl/utils/bert_padding.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+# limitations under the License.
+# reference from Dao-AILAB flash-attn
+# https://github.com/Dao-AILab/flash-attention/blob/74b0761ff7efc7b90d4e5aeb529c1b2a09a7458c/flash_attn/bert_padding.py#L38
+import paddle
+import paddle.nn.functional as F
+
+from paddlenlp.utils.log import logger
+
+try:
+    from einops import rearrange, repeat
```
Comment (gongel) [inline paddlenlp/rl/utils/bert_padding.py:20]: 这里需要一个安装的依赖，加到 requirements 里面

**Review by gongel — COMMENTED** [review]
1. 缺一个精度验证，可以挂个实验跑起来。
2. 缺少文档，能够跑起这些优化

**Review by gongel — APPROVED** [review]
